pkg/slo-controller/noderesource/plugins/gpudeviceresource/plugin.go

```diff
@@ -149,26 +151,39 @@ func (p *Plugin) Calculate(_ *configuration.ColocationStrategy, node *corev1.Nod
 		return p.resetGPUNodeResource()
 	}
 
+	enableSyncGPUSharedResource := false
+	if utilfeature.DefaultFeatureGate.Enabled(features.EnableSyncGPUSharedResource) {
+		enableSyncGPUSharedResource = true
+	}
+
 	// TODO: calculate NUMA-level resources against NRT
-	return p.calculate(node, device)
+	return p.calculate(enableSyncGPUSharedResource, node, device)
 }
 
-func (p *Plugin) calculate(node *corev1.Node, device *schedulingv1alpha1.Device) ([]framework.ResourceItem, error) {
+func (p *Plugin) calculate(enableSyncGPUSharedResource bool, node *corev1.Node, device *schedulingv1alpha1.Device) ([]framework.ResourceItem, error) {
 	if device == nil {
 		return nil, fmt.Errorf("invalid device")
 	}
 
 	// calculate gpu resources
 	gpuResources := make(corev1.ResourceList)
 	totalKoordGPU := resource.NewQuantity(0, resource.DecimalSI)
+	healthGPUNum := 0
 	for _, d := range device.Spec.Devices {
 		if d.Type != schedulingv1alpha1.GPU || !d.Health {
 			continue
 		}
+
+		healthGPUNum++
 		util.AddResourceList(gpuResources, d.Resources)
 		totalKoordGPU.Add(d.Resources[extension.ResourceGPUCore])
 	}
+
 	gpuResources[extension.ResourceGPU] = *totalKoordGPU
+	if enableSyncGPUSharedResource {
```

Comment:
why not just `if utilfeature.DefaultFeatureGate.Enabled(features.EnableSyncGPUSharedResource)` {
  gpuResources[extension.ResourceGPUShared] = *resource.NewQuantity(int64(healthGPUNum)*100, resource.DecimalSI)
	}
}

Reply:
fixed